src/model/submissions.nit

```diff
@@ -0,0 +1,202 @@
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# Player's submission for Pep8 missions
+#
+# TODO: Pep8Term is crap software to install and to execute.
+# Currently, this program assume the following:
+#
+# * `pep8` and `asem8` are in the PATH
+# * `trap` and `pep8os.pepo` are in the current working directory
+module submissions
+
+import missions
+import players
+import markdown
+import poset
+
+redef class TestCase
+	# Try the test case on a `program`.
+	fun run(program: Program, i: Int): TestResult do
+		var res = new TestResult(self, program)
+
+		# We get a subdirectory (a testspace) for each test case
+		var ws = program.workspace.as(not null)
+		var tdir = "test{i}"
+		var ts = ws / tdir
+		ts.mkdir
+
+		# Prepare the input/output
+		var ifile = ts / "input.txt"
+		self.i.write_to_file(ifile)
+		var ofile = ts / "output.txt"
+		var sfile = ts / "sav.txt"
+		self.o.write_to_file(sfile)
+
+		# Prepare the execution command
+		# Because `pep8` is interactive.
+		var canned_command = """
+l
+source
+i
+f
+{{{tdir}}}/input.txt
+o
+f
+{{{tdir}}}/output.txt
+x
+q
+"""
+		canned_command.write_to_file(ts/"canned_command")
+
+		# Try to execute the program on the test input
+		# TODO: some time/space limit!
```

Comment:
We could assume that, the main issue I see is denial of service by time or output exhaustion.